### PR TITLE
Move all uses of Core Foundation to be @_implementationOnly except getCFRunLoop().

### PR DESCRIPTION
**Pull request by millenomi**
The CF headers and module are still required (because of getCFRunLoop()), but this makes them much easier to remove in the future.

### CODE REVIEW

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
cc @compnerd

**Comment by millenomi** [comment]
Test failure looks unrelated.

**Comment by millenomi** [comment]
@swift-ci please test linux

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
It was not unrelated.

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
@swift-ci please test

**Comment by millenomi** [comment]
Tests appear to be broken regardless of this patch. Pulling the PR brake.

File: Sources/Foundation/NSCFTypeShims.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
+//
+//  NSCFTypeShims.swift
+//  SwiftFoundation
+//
+//  Created by Aura Lily Vulcano on 8/27/20.
+//  Copyright © 2020 Swift. All rights reserved.
```
Comment (xwu) [inline Sources/Foundation/NSCFTypeShims.swift:6]: Nit: Swift project header :)

**Pull request merged**